src/variant_detection/variant_output.cpp

```diff
@@ -1,28 +1,104 @@
 #include "variant_detection/variant_output.hpp"
 
+#include <chrono>   // for std::chrono::system_clock
+#include <ctime>    // for std::localtime, std::time, std::time_t
 #include <iostream> // for std::cout
+#include <iomanip>  // for std::put_time
 
-#include "structures/junction.hpp"              // for class Junction
-#include "variant_parser/variant_record.hpp"    // for class variant_header
+#include <bio/var_io/writer.hpp>        // for bio::var_io::header
 
-void find_and_output_variants(std::map<std::string, int32_t> & references_lengths,
-                              std::vector<Cluster> const & clusters,
-                              cmd_arguments const & args,
-                              std::ostream & out_stream)
+#include <seqan3/core/debug_stream.hpp> // for seqan3::debug_stream
+
+using namespace std::string_literals;
+using namespace seqan3::literals;
+
+std::string transTime()
+{
+    std::stringstream text_stream;
+    const std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now();
+    const std::time_t rawtime = std::chrono::system_clock::to_time_t(now);
+    text_stream << std::put_time(std::localtime(&rawtime), "%F %T");
+    std::string time_string = text_stream.str();
+    return time_string;
+}
+
+/*! \brief Writes a VCF header.
+ *
+ * \param[in] references_lengths - reference sequence dictionary parsed from \@SQ header lines
+ * \param[in] sample_name        - Name of the sample for the vcf header line
+ * \param[in, out] hdr           - reference sequence dictionary parsed from \@SQ header lines
+ */
+void write_header(std::map<std::string, int32_t> & references_lengths,
+                  std::string sample_name,
+                  bio::var_io::header & hdr)
+{
+    hdr.file_format = "VCFv4.3";
+    // Default:
+    // hdr.filters.push_back({ .id = "PASS", .description = "All filters passed"});
+    hdr.infos.push_back({ .id = "END", .number = 1, .type = bio::var_io::value_type_id::int32,
+                          .description = "End position of SV called."});
+    hdr.infos.push_back({ .id = "SVLEN", .number = 1, .type = bio::var_io::value_type_id::int32,
+                          .description = "Length of SV called."});
+    hdr.infos.push_back({ .id = "SVTYPE", .number = 1, .type = bio::var_io::value_type_id::string,
+                          .description = "Type of SV called."});
+    hdr.formats.push_back({ .id = "GT", .number = 1, .type = bio::var_io::value_type_id::string,
+                          .description = "Genotype"});
+
+    for (auto const & [id, length] : references_lengths)
+        hdr.contigs.push_back({ .id = id, .length = length});
+
+    // std::string filedate = "filedate=", transTime(),
+    hdr.other_lines = {"filedate=" + transTime(),
```

Comment:
Does this work? I think you might need to add an s:
```suggestion
    hdr.other_lines = {"filedate="s + transTime(),
```
?

Reply:
Works both